src/server/auth-token.js

```diff
@@ -61,21 +61,22 @@ class AuthToken {
   static authenticate(req, res, next) {
     const bearerToken = req.headers.authorization;
     if (!bearerToken) {
-      return next();
+      next();
+      return;
     }
     const jwtToken = bearerToken.split(' ')[1];
-    return AuthToken.decode(TOKEN_TYPE.ACCESS, jwtToken)
-      .then(payload => {
-        req.headers.userKey = payload.userKey;
-        req.headers.permission = USER_PERMISSION[payload.userType];
-        next();
-      })
-      .catch(err => {
-        next(new APIError(err, {
-          statusCode: 400,
-          message: err.message
-        }));
-      });
+    AuthToken.decode(TOKEN_TYPE.ACCESS, jwtToken)
```

Comment:
return is not required. next() calls the next api function assigned in the router, but return can be used.